Make SQLite use libpas for malloc in NetworkProcess

bnham opened the pull request:
#### 4a3c8a48a35408877f222cf3dab26901a9c7b887
<pre>
Make SQLite use libpas for malloc in NetworkProcess
<a href="https://bugs.webkit.org/show_bug.cgi?id=243642">https://bugs.webkit.org/show_bug.cgi?id=243642</a>
&lt;rdar://problem/98288137&gt;

Reviewed by Yusuke Suzuki.

This makes SQLite use libpas instead of system malloc in NetworkProcess. It seems to be a ~0.5% win
in Membuster. The wins come from:

1. libpas being more aggressive at returning memory to the OS via MADV_FREE.
2. libpas being fast enough that we can disable the lookaside allocator, which was dirtying 128KB of
   memory per connection. I tested this by running various IDB benchmarks which seemed neutral after
   this making this change.

This requires exporting libpas&apos;s implementation of malloc_size to WTF, since the current
implementation of that function in WTF doesn&apos;t do anything.

* Source/WTF/wtf/FastMalloc.cpp:
(WTF::fastMallocSize):
* Source/WebCore/platform/sql/SQLiteDatabase.cpp:
(WebCore::SQLiteDatabase::useFastMalloc):
* Source/WebCore/platform/sql/SQLiteDatabase.h:
* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
(WebKit::NetworkProcess::initializeNetworkProcess):
* Source/bmalloc/bmalloc/bmalloc.cpp:
(bmalloc::api::mallocSize):
* Source/bmalloc/bmalloc/bmalloc.h:

Canonical link: <a href="https://commits.webkit.org/253215@main">https://commits.webkit.org/253215@main</a>
</pre>

Constellation commented on Source/WebCore/platform/sql/SQLiteDatabase.cpp:
```diff
@@ -76,6 +83,28 @@ static void initializeSQLiteIfNecessary()
 static Lock isDatabaseOpeningForbiddenLock;
 static bool isDatabaseOpeningForbidden WTF_GUARDED_BY_LOCK(isDatabaseOpeningForbiddenLock) { false };
 
+void SQLiteDatabase::useFastMalloc(void)
```
`void` in parameters are not necessary.

Constellation commented on Source/WebCore/platform/sql/SQLiteDatabase.h:
```diff
@@ -157,6 +157,8 @@ class SQLiteDatabase : public CanMakeWeakPtr<SQLiteDatabase> {
     void disableThreadingChecks() { }
 #endif
 
+    WEBCORE_EXPORT static void useFastMalloc(void);
```
Ditto.

Constellation reviewed (approved):
r=me, nice!

bnham committed:
Make SQLite use libpas for malloc in NetworkProcess
https://bugs.webkit.org/show_bug.cgi?id=243642
<rdar://problem/98288137>

Reviewed by Yusuke Suzuki.

This makes SQLite use libpas instead of system malloc in NetworkProcess. It seems to be a ~0.5% win
in Membuster. The wins come from:

1. libpas being more aggressive at returning memory to the OS via MADV_FREE.
2. libpas being fast enough that we can disable the lookaside allocator, which was dirtying 128KB of
   memory per connection. I tested this by running various IDB benchmarks which seemed neutral after
   this making this change.

This requires exporting libpas's implementation of malloc_size to WTF, since the current
implementation of that function in WTF doesn't do anything.

* Source/WTF/wtf/FastMalloc.cpp:
(WTF::fastMallocSize):
* Source/WebCore/platform/sql/SQLiteDatabase.cpp:
(WebCore::SQLiteDatabase::useFastMalloc):
* Source/WebCore/platform/sql/SQLiteDatabase.h:
* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
(WebKit::NetworkProcess::initializeNetworkProcess):
* Source/bmalloc/bmalloc/bmalloc.cpp:
(bmalloc::api::mallocSize):
* Source/bmalloc/bmalloc/bmalloc.h:

Canonical link: https://commits.webkit.org/253215@main
